src/tools/cephfs_mirror/Mirror.cc

```diff
@@ -496,31 +494,33 @@ void Mirror::update_fs_mirrors() {
   dout(20) << dendl;
 
   auto now = ceph_clock_now();
+
   double blocklist_interval = g_ceph_context->_conf.get_val<std::chrono::seconds>
     ("cephfs_mirror_restart_mirror_on_blocklist_interval").count();
-  bool check_blocklist = blocklist_interval > 0 && ((now - m_last_blocklist_check) >= blocklist_interval);
-
   double failed_interval = g_ceph_context->_conf.get_val<std::chrono::seconds>
     ("cephfs_mirror_restart_mirror_on_failure_interval").count();
-  bool check_failure = failed_interval > 0 && ((now - m_last_failure_check) >= failed_interval);
 
   {
     std::scoped_lock locker(m_lock);
     for (auto &[filesystem, mirror_action] : m_mirror_actions) {
       auto failed = mirror_action.fs_mirror && mirror_action.fs_mirror->is_failed();
       auto blocklisted = mirror_action.fs_mirror && mirror_action.fs_mirror->is_blocklisted();
 
-      if (check_failure && !mirror_action.action_in_progress &&
-	  !_is_restarting(filesystem) && failed) {
+      if (failed &&
+	  !_is_restarting(filesystem) &&
+	  !mirror_action.action_in_progress &&
+	  (failed_interval > 0 && (mirror_action.fs_mirror->get_failed_ts() - now) > failed_interval)) {
         // about to restart failed mirror instance -- nothing
         // should interfere
         dout(5) << ": filesystem=" << filesystem << " failed mirroring -- restarting" << dendl;
 	_set_restarting(filesystem);
         auto peers = mirror_action.fs_mirror->get_peers();
         auto ctx =  new C_RestartMirroring(this, filesystem, mirror_action.pool_id, peers);
         ctx->complete(0);
-      } else if (check_blocklist && !mirror_action.action_in_progress &&
-		 !_is_restarting(filesystem) && blocklisted) {
+      } else if (blocklisted &&
+		 !_is_restarting(filesystem) &&
+		 !mirror_action.action_in_progress &&
+		 (blocklist_interval > 0 && (mirror_action.fs_mirror->get_blocklisted_ts() - now) > blocklist_interval)) {
```

Comment:
here too, maybe add a variable `bool check_blocklist` and assign this long line to it